smp/smp.c

```diff
@@ -94,13 +94,13 @@ static __text_init void boot_cpu(unsigned int cpu) {
 }
 
 void __text_init init_smp(void) {
-    unsigned mp_nr_cpus = mptables_get_nr_cpus();
-    unsigned acpi_nr_cpus = acpi_get_nr_cpus();
-
-    nr_cpus = acpi_nr_cpus ?: mp_nr_cpus;
+    nr_cpus = acpi_get_nr_cpus();
     if (nr_cpus == 0) {
-        nr_cpus++;
-        return;
+        nr_cpus = mptables_get_nr_cpus();
+        if (nr_cpus == 0) {
```

Comment:
If mptables_get_nr_cpus returns 0, then why do we increment this by 1?

Reply:
looks like this is existing code. 
Does 0 means atleast 1 cpu is present?

Reply:
It means that neither ACPI MADT nor MP Tables are available. So we assume that there is just the current one BSP.